luigi/contrib/hadoop.py

```diff
@@ -523,7 +524,10 @@ def run_job(self, job, tracking_url_callback=None):
 
         job.dump(self.tmp_dir)
 
-        run_and_track_hadoop_job(arglist, tracking_url_callback=tracking_url_callback)
+        job_env = os.environ.copy()
+        if self.username:
+            job_env["HADOOP_USER_NAME"] = self.username
```

Comment:
the whole environment dict is passed to `run_and_track_hadoop_job` so it looks like it's fine